pages/web-literacy/WebLiteracyPage.jsx

```diff
@@ -1,44 +1,314 @@
+      </Illustration>
+    );
+  }
+});
+
+var WebLitPage = React.createClass({
+  mixins: [History],
   statics: {
     pageTitle: "Web Literacy",
     pageClassName: "web-literacy"
   },
-  render: function() {
-    var whitepaperLink = "https://mozilla.github.io/webmaker-whitepaper";
-    return (
-      <div className="inner-container">
-        <h1>Web Literacy</h1>
+  updateMapNavState: function() {
+    var topic = this.props.params.verb || "";
+    var webLitSkill = this.props.params.webLitSkill || "";
+    var state = {};
+    if (this.state.topic !== topic || this.state.webLitSkill !== webLitSkill) {
+      state.topic = topic;
+      state.webLitSkill = webLitSkill;
+      this.setState(state);
+    }
+  },
+  componentDidUpdate: function() {
+    this.updateMapNavState();
+  },
+  componentDidMount: function() {
+    this.updateMapNavState();
+  },
+  getInitialState: function() {
+    var filter = {};
+    Object.keys(categories).map(function(cat) {
+      filter[categories[cat]] = false;
+    });
+    return {
+      topic: "",
+      webLitSkill: "",
+      filter: filter
+    };
+  },
+  hasWebLitSkillIn: function(webLitSkills) {
+    var webLitSkill = this.state.webLitSkill;
+    if (webLitSkill && webLitSkills.indexOf(webLitSkill) !== -1) {
+      return true;
+    }
+  },
+  hasMatchingWebLitSkillIn: function(webLitSkills) {
+    var selectedTopic = this.state.topic;
+    if (!selectedTopic || this.state.webLitSkill) {
+      return false;
+    }
+    var topicWebLitSkills = Object.keys(weblitdata[selectedTopic]);
+    return webLitSkills.some(function(webLitSkill) {
+      return topicWebLitSkills.indexOf(webLitSkill) !== -1;
+    });
+  },
+  hasMatching21CSkillIn: function(skills21C) {
+    var state = this.state;
+    var hasCategory = this.hasCategory;
+    return skills21C.some(function(skill21C) {
+      return !state.filter[categories[skill21C]] && hasCategory(skill21C, state.topic, state.webLitSkill);
+    });
+  },
+  hasCategory: function(cat, selectedVerb, selectedWebLitSkill) {
+    var cat = categories[cat];
+
+    if (!selectedVerb) {
+      return true;
+    }
 
-        <section className="intro">
+    if (!selectedWebLitSkill) {
+      return Object.keys(weblitdata[selectedVerb]).some(function(item) {
+        return weblitdata[selectedVerb][item].indexOf(cat) !== -1;
+      });
+    }
+
+    return weblitdata[selectedVerb][selectedWebLitSkill].indexOf(cat) !== -1;
+  },
+  renderActivities: function() {
+    var activities = [];
+    activitydata.forEach(function(activity, index) {
+      if ((this.hasWebLitSkillIn(activity.webLitSkills) || this.hasMatchingWebLitSkillIn(activity.webLitSkills)) && this.hasMatching21CSkillIn(activity.skills)) {
+        activities.push(
+          <div className="activity-item" key={index}>
+            <Activity
+              selectedTopic={this.state.topic}
+              selectedWebLitSKill={this.state.webLitSkill}
+              topic={activity.topic}
```

Comment:
feels like we should do
```
activity.src1x = activity.imgSrc1x;
activity.src2x = activity.imgSrc2x;
...
<Activity selectedTopic={this.state.topic} selectedWebLitSKill={this.state.webLitSkill} {...activity}/>
```
where `{...activity}` is ES6 notation but is also part of JSX, so works even without ES6->ES5 conversions